### PR TITLE
Task2

### PR DESCRIPTION
**Pull request by uladzimirzhyhar**

### CODE REVIEW

File: script/task2.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
+'use strict';
+
+function isArray(array) {
+  return Array.isArray(array);
```
Comment (NikitaSedyx) [inline script/task2.js]: how do you think is there sense to wrap only function to get the same result, imo No

isArray2, isArray3 are fine

File: script/task2.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+console.log('isArray3 result:');
+console.log(isArray3([1, 3, 4]));
+console.log(isArray3(11));
+console.log(isArray3({}));
+console.log(isArray3(undefined));
+console.log(isArray3(NaN));
+console.log(isArray3([]));
+
+function range(start, end, step) {
```
Comment (NikitaSedyx) [inline script/task2.js]:
```
10 Array [ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 ]
1 10 Array [ 1, 2, 3, 4, 5, 6, 7, 8, 9 ]
1 10 3 Array [ 1, 4, 7 ]
10 null 3 Array [  ]
10 null Array [0..9]
-2 -5 Array [  ]
-5 -2 Array [ -5, -4, -3 ]
-5 -2 2 Array [ -5, -3 ]
-5 -2 -2 Array [  ]
-5 null 2 Array [ ]
-5 null -2 Array [ 0, -2, -4 ]
-10 -20 -5 Array [ -10, -15 ]
-20 -10 -5 Array [  ]
```
you should receive the same result

Comment (uladzimirzhyhar) [inline script/task2.js]: 10 null 3 Array [  ] - this test should return [0, 3, 6, 9]

Comment (NikitaSedyx) [inline script/task2.js]: it should return [], from 10 to null with iterator 3, it's impossible

File: script/task2.js [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    end = start;
+    start = 0;
+  }
+  
+  if (step == undefined) {
+    step = Math.abs(end) - start > 0 ? 1 : -1;
+  }
+  
+  if (step > 0) {
+    for (var i = start; i < end; i+=step) {
```
Comment (NikitaSedyx) [inline script/task2.js]: it could be 1 loop

**Pull request merged**